src/container.cpp

```diff
@@ -383,15 +381,15 @@ ReturnValue Container::queryMaxCount(int32_t index, const Thing& thing, uint32_t
 	return RETURNVALUE_NOERROR;
 }
 
-ReturnValue Container::queryRemove(const Thing& thing, uint32_t count, uint32_t flags,
-                                   Creature* actor /*= nullptr */) const
+ReturnValue Container::queryRemove(std::shared_ptr<const Thing> thing, uint32_t count, uint32_t flags,
+                                   std::shared_ptr<Creature> actor /*= nullptr */) const
 {
-	int32_t index = getThingIndex(&thing);
+	int32_t index = getThingIndex(thing);
 	if (index == -1) {
 		return RETURNVALUE_NOTPOSSIBLE;
 	}
 
-	const Item* item = thing.getItem();
+	auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
	const auto item = thing->getItem();
```